tokio/src/sync/rwlock.rs

```diff
@@ -199,8 +202,35 @@ impl<T: ?Sized> RwLock<T> {
         T: Sized,
     {
         RwLock {
+            mr: MAX_READS,
             c: UnsafeCell::new(value),
-            s: Semaphore::new(MAX_READS),
+            s: Semaphore::new(MAX_READS as usize),
+        }
+    }
+
+    /// Creates a new instance of an `RwLock<T>` which is unlocked
+    /// and allows a maximum of `max_reads` concurrent readers.
+    ///
+    /// # Examples
+    ///
```

Comment:
This should include a section on panics.

Reply:
Is it worth making the `MAX_READS` const public just so we can link to it, or we can just mention the exact number in the docs ?

Reply:
We don't make any of the other constants public. You can mention it as `u32::MAX >> 3` or `2^29` as you prefer.